src/fondant/core/component_spec.py

```diff
@@ -313,12 +294,12 @@ class OperationSpec:
 
     def __init__(
         self,
-        specification: ComponentSpec,
+        component_spec: ComponentSpec,
         *,
         consumes: t.Optional[t.Dict[str, t.Union[str, pa.DataType]]] = None,
         produces: t.Optional[t.Dict[str, t.Union[str, pa.DataType]]] = None,
     ) -> None:
-        self.specification = specification
+        self.component_spec = component_spec
```

Comment:
```suggestion
        self._component_spec = component_spec
```

Can we make this private and expose all the attributes that are used publicly as properties? That way this class is more in control of how it's used, and linters will complain if the class is used incorrectly.

Reply:
Make sense :) added the changes, now the component spec is completely hidden from the executor